public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -83,6 +93,11 @@ export class AppChromeService {
     }
   }
 
+  public setReturnToPrevious(returnToPrevious: ReturnToPreviousProps) {
+    this.update({ returnToPrevious });
+    window.sessionStorage.setItem('returnToPrevious', JSON.stringify(returnToPrevious));
+  }
+
```

Comment:
can we have a `clearReturnToPrevious` method:
```tsx
public clearReturnToPrevious() {
  this.update({ returnToPrevious: undefined });
  window.sessionStorage.removeItem('returnToPrevious');
}
```

then you can call that from the `ReturnToPrevious` component in `handleOnClick` and `handleDismiss`